scripts/sec/sec.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+# Copyright 2023 OpenHW Group
+#
+# Licensed under the Solderpad Hardware Licence, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://solderpad.org/licenses/
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+usage() {                                 # Function: Print a help message.
+  echo "Usage: $0 [ -t {cadence,synopsys,mentor} ]" 1>&2
+}
+exit_abnormal() {                         # Function: Exit with error.
+  usage
+  exit 1
+}
+
+while getopts "t:" flag
+do
+    case "${flag}" in
+        t)
+        target_tool=${OPTARG}
+        ;;
+        :)
+        exit_abnormal
+        ;;
+        *)
+        exit_abnormal
+        ;;
+        ?)
+        exit_abnormal
+        ;;
+    esac
+done
+
+if [ ! -d ./reports/ ]; then
+    mkdir -p ./reports/
+fi
+
+if [[ "${target_tool}" != "cadence" && "${target_tool}" != "synopsys" && "${target_tool}" != "mentor" ]]; then
+    exit_abnormal
+fi
+
+if [[ -z "${GOLDEN_RTL}" ]]; then
+    echo "The env variable GOLDEN_RTL is empty."
+    if [ ! -d "./ref_design" ]; then
+        echo "Cloning Golden Design...."
+        git clone https://github.com/openhwgroup/cve2.git ref_design
+    fi
+    export GOLDEN_RTL=$(pwd)/ref_design/rtl
+else
+    echo "SEC: Using ${GOLDEN_RTL} as reference design"
+fi
+
+REVISED_DIR=$( readlink -f $(pwd)/../../)
+
+GOLDEN_DIR=$(readlink -f ./ref_design/)
+
+
+var_golden_rtl=$(awk '{ if ($0 ~ "{DESIGN_RTL_DIR}") print $0 }' ${GOLDEN_DIR}/cv32e20_manifest.flist | sed 's|${DESIGN_RTL_DIR}|./ref_design/rtl/|')
```

Comment:
A comment in .flist file might leak in to the var_*_rtl variable and cause ignoring the remainder of the command line options of jg, as "#" sign is interpreted that way.

Similarly, in the original sh file (SEC script from e40) the project files were filtered out off "wrapper" and "tracer". Why are they kept here?

Reply:
For the first it only includes in the string the `{DESIGN_RTL_DIR}` lines so if a comment has it it would be included, probably I could secure it a little more not allowing this characters in the string.

Regarding the wrapper and tracer I can also filter them if you want :)

Reply:
Regarding filtering-out "tracer" and "wrapper", it would probably be ignored by jg anyway, as the -spec_top and -imp_top already specifies the top module. Therefore, it might not be necessary to filter it. I'll leave it up to you to keep it as in the original script, or use the logic of jg to ignore the irrelevant code.

Reply:
As we don't even use it for the checks I think it is better to exclude it. But it is not a strong opinion :smile: